src/terrain/draw_terrain_raster.js

```diff
@@ -174,41 +192,191 @@ function drawTerrainRaster(painter: Painter, terrain: Terrain, sourceCache: Sour
+            program = painter.useProgram('terrainRaster', null, modes);
+            programMode = mode;
+        };
+
+        const colorMode = painter.colorModeForRenderPass();
+        const depthMode = new DepthMode(gl.LEQUAL, DepthMode.ReadWrite, painter.depthRangeFor3D);
+        vertexMorphing.update(now);
+        const tr = painter.transform;
+        const skirt = skirtHeight(tr.zoom) * terrain.exaggeration();
+
+        const batches = showWireframe ? [false, true] : [false];
+
+        batches.forEach(isWireframe => {
+            // This code assumes the rendering is batched into mesh terrain and then wireframe
+            // terrain (if applicable) so that this is enough to ensure the correct program is
+            // set when we switch from one to the other.
+            programMode = -1;
+
+            const primitive = isWireframe ? gl.LINES : gl.TRIANGLES;
+            const [buffer, segments] = isWireframe ? terrain.getWirefameBuffer() : [terrain.gridIndexBuffer, terrain.gridSegments];
+
+            for (const coord of tileIDs) {
+                const tile = sourceCache.getTile(coord);
+                const stencilMode = StencilMode.disabled;
+
+                const prevDemTile = terrain.prevTerrainTileForTile[coord.key];
+                const nextDemTile = terrain.terrainTileForTile[coord.key];
+
+                if (demTileChanged(prevDemTile, nextDemTile)) {
+                    vertexMorphing.newMorphing(coord.key, prevDemTile, nextDemTile, now, defaultDuration);
+                }
+
+                // Bind the main draped texture
+                context.activeTexture.set(gl.TEXTURE0);
+                tile.texture.bind(gl.LINEAR, gl.CLAMP_TO_EDGE, gl.LINEAR_MIPMAP_NEAREST);
+
+                const morph = vertexMorphing.getMorphValuesForProxy(coord.key);
+                const shaderMode = morph ? SHADER_MORPHING : SHADER_DEFAULT;
+                let elevationOptions;
+
+                if (morph) {
+                    elevationOptions = {morphing: {srcDemTile: morph.from, dstDemTile: morph.to, phase: easeCubicInOut(morph.phase)}};
+                }
+
+                const uniformValues = terrainRasterUniformValues(coord.projMatrix, isEdgeTile(coord.canonical, tr.renderWorldCopies) ? skirt / 10 : skirt);
+                setShaderMode(shaderMode, isWireframe);
+
+                terrain.setupElevationDraw(tile, program, elevationOptions);
+
+                painter.prepareDrawProgram(context, program, coord.toUnwrapped());
+
+                program.draw(context, primitive, depthMode, stencilMode, colorMode, CullFaceMode.backCCW,
+                    uniformValues, "terrain_raster", terrain.gridBuffer, buffer, segments);
+            }
+        });
+    }
+}
+
 function drawTerrainDepth(painter: Painter, terrain: Terrain, sourceCache: SourceCache, tileIDs: Array<OverscaledTileID>) {
-    assert(painter.renderPass === 'offscreen');
+    if (painter.transform.projection.name === 'globe') {
```

Comment:
TODO: Skip depth rendering for globe, and use a 1x1 texture object placeholder for symbols.

Reply:
Addressed in https://github.com/mapbox/mapbox-gl-js/pull/11329/commits/34422a659c97044368a04571dca03961625eae89